compilers/concrete-compiler/compiler/lib/Common/Transformers.cpp

```diff
@@ -281,12 +281,36 @@ Result<TransportValueVerifier> getTransportValueVerifier(
       return StringError(
           "Tried to transform a transport value without type infos.");
     }
-    if ((capnp::AnyStruct::Reader)gateInfo.asReader().getTypeInfo() !=
-        (capnp::AnyStruct::Reader)transportVal.asReader().getTypeInfo()) {
+    // REMOVE ME: Hacking variance comparison with an epsilon as it can mismatch
+    // between the RO and PO
+    auto cmpVal = transportVal;
+    if (gateInfo.asReader().getTypeInfo().hasLweCiphertext() &&
+        transportVal.asReader().getTypeInfo().hasLweCiphertext()) {
+      auto gateVariance = gateInfo.asReader()
+                              .getTypeInfo()
+                              .getLweCiphertext()
+                              .getEncryption()
+                              .getVariance();
+      auto transportVariance = transportVal.asReader()
+                                   .getTypeInfo()
+                                   .getLweCiphertext()
+                                   .getEncryption()
+                                   .getVariance();
+      if (std::abs(gateVariance - transportVariance) > 1e-40) {
+        return StringError("Tried to transform a transport value with "
+                           "incompatible variance.");
+      }
+      cmpVal.asBuilder()
+          .getTypeInfo()
+          .getLweCiphertext()
+          .getEncryption()
+          .setVariance(gateVariance);
+    } else if ((capnp::AnyStruct::Reader)gateInfo.asReader().getTypeInfo() !=
+               (capnp::AnyStruct::Reader)cmpVal.asReader().getTypeInfo()) {
```

Comment:
I don't think it's an `else/if` that we need here, isn't it? We should only update the variance if needed and leave the flow as it was before